The job **`dist-i686-mingw`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/5170141697/jobs/9312856334) [(plain)](https://github.com/rust-lang-ci/rust/commit/c3121adc42eb406699a89833587bee3f4beff00c/checks/13990494598/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling tidy v0.1.0 (C:\a\rust\rust\src\tools\tidy)
[RUSTC-TIMING] tidy test:false 5.311
[RUSTC-TIMING] cargo_metadata test:false 7.479
   Compiling unstable-book-gen v0.1.0 (C:\a\rust\rust\src\tools\unstable-book-gen)
error: linking with `i686-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "i686-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-Wl,--large-address-aware" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsbegin.o" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcoWXBb9\\symbols.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.0.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.1.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.10.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.11.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.12.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.13.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.14.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.15.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.2.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.3.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.4.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.5.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.6.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.7.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.8.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.unstable_book_gen.24981240f60c3bc5-cgu.9.rcgu.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.2l5sbsinm05jchqc.rcgu.o" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\release\\deps" "-L" "C:\\Users\\runneradmin\\.cargo\\registry\\src\\index.crates.io-1cd66030c949c28d\\winapi-i686-pc-windows-gnu-0.4.0\\lib" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libtidy-deee54c4e172370e.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libmiropt_test_tools-102501b8e07ea5a1.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libignore-a3b01ed3f4d2f7fe.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\liblazy_static-d3037c224c0074c5.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libwalkdir-ee275cec02ba48d4.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libsame_file-495aef4ab09f1831.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libthread_local-f24b1e026df6a7cd.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libonce_cell-cf7f5ac641f23725.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libglobset-10a96eb7f1f89b46.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libfnv-f0181ec8e617987f.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\liblog-371885e997c48e85.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libcfg_if-9739b5659e918827.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libbstr-df37efc8271d03ee.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libregex-2243ddbe483c5fbf.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libaho_corasick-170e0be632aaec5d.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libmemchr-c510e3f14e967ebd.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libregex_syntax-18631fd8920af995.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libcargo_metadata-b84e7341b25be9f8.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libserde_json-600b365037b6dcbb.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libryu-8786f0c2ff1957bb.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libitoa-3c6d3ba379cf3772.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libthiserror-5f9c8fb13dc09910.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libcargo_platform-61fa167ded5e5057.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libsemver-d4f240716ef4e26a.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libcamino-6942dfb7dc136523.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libserde-ea83c0805f78c14b.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libtermcolor-ff73011b23995e38.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libwinapi_util-e55d159aac6151c6.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\libwinapi-eeba17fc2de47fc4.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libstd-7601ac96d9612508.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libpanic_unwind-d81935824487db4d.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libobject-14d9e23b92ea7ddd.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libmemchr-b7b66cf9911f1aef.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libaddr2line-27b3808a7d9c4d74.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libgimli-0ac5c7376aa46619.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_demangle-40991a5515a1aec5.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libstd_detect-5467522f3dd1675a.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libhashbrown-1792159f4e67ed66.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_std_workspace_alloc-822d8e132b27c8ef.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libminiz_oxide-cacaf4244265885b.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libadler-0c1701755722f143.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libunwind-393c9009a850229c.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcfg_if-29e98cc9d5aabdf6.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\liblibc-82ceaaa9f1482620.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\liballoc-d2771c336cb1eca7.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_std_workspace_core-3814a240adea6185.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcore-a3d4185eeefda4f6.rlib" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcompiler_builtins-5c0d92b338a1c328.rlib" "-Wl,-Bdynamic" "-lwinapi_advapi32" "-lwinapi_cfgmgr32" "-lwinapi_gdi32" "-lwinapi_kernel32" "-lwinapi_msimg32" "-lwinapi_opengl32" "-lwinapi_user32" "-lwinapi_winspool" "-lkernel32" "-ladvapi32" "-lbcrypt" "-lkernel32" "-lntdll" "-luserenv" "-lws2_32" "-lkernel32" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0-bootstrap-tools\\i686-pc-windows-gnu\\release\\deps\\unstable_book_gen-66bdaa7dfe7fbcb4.exe" "-Wl,--gc-sections" "-no-pie" "-Wl,-O1" "-nodefaultlibs" "-gz" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage0\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsend.o"
  = note: C:/a/rust/rust/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: unrecognized option '--compress-debug-sections=zlib'
          C:/a/rust/rust/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: use the --help option for usage information

          collect2.exe: error: ld returned 1 exit status
          
          

[RUSTC-TIMING] unstable_book_gen test:false 1.188
error: could not compile `unstable-book-gen` (bin "unstable-book-gen") due to previous error
```

</details>

--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -1673,7 +1673,15 @@ impl<'a> Builder<'a> {
                 self.config.rust_debuginfo_level_tools
             }
         };
-        cargo.env(profile_var("DEBUG"), debuginfo_level.to_string());
+        if debuginfo_level == 1 {
+            // Use less debuginfo than the default to save on disk space.
+            cargo.env(profile_var("DEBUG"), "line-tables-only");
+        } else {
+            cargo.env(profile_var("DEBUG"), debuginfo_level.to_string());
+        };
+        if self.cc[&target].args().iter().any(|arg| arg == "-gz") {
+            rustflags.arg("-Clink-arg=-gz");
+        }
         cargo.env(
             profile_var("DEBUG_ASSERTIONS"),
             if mode == Mode::Std {

--- a/src/bootstrap/cc_detect.rs
+++ b/src/bootstrap/cc_detect.rs
@@ -69,6 +69,8 @@ fn new_cc_build(build: &Build, target: TargetSelection) -> cc::Build {
         .opt_level(2)
         .warnings(false)
         .debug(false)
+        // Compress debuginfo
+        .flag_if_supported("-gz")
         .target(&target.triple)
         .host(&build.build.triple);
     match build.crt_static(target) {

